### PR TITLE
add send mail action with success and failure notifications

### PR DESCRIPTION
**Pull request by patrickweh**
## Summary by Sourcery

Extract email sending into a new SendMail action, refactor the Livewire EditMail component to use this action with success/failure notifications, and add comprehensive unit tests for various mailing scenarios

New Features:
- Introduce SendMail action to centralize email sending with template rendering, attachments, and optional queuing
- Add success, failure, and mixed-result notifications in the EditMail component based on email send outcomes
- Enable persistent behavior on the edit-mail modal

Enhancements:
- Refactor EditMail send method to use SendMail action instead of direct Mail facade calls

Tests:
- Add unit tests for SendMail covering failures, null handling, string and array addresses, template overrides, queued sends, attachments, blade parameters, CC/BCC, and default template values

Chores:
- Add SendMailRuleset for validation of mail payloads

### CODE REVIEW

File: src/Rulesets/MailMessage/SendMailRuleset.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+{
+    public function rules(): array
+    {
+        return [
+            'to' => 'required|array',
+            'to.*' => 'email',
+            'cc' => 'nullable|array',
+            'cc.*' => 'email',
+            'bcc' => 'nullable|array',
+            'bcc.*' => 'email',
```
[the remark this answers is not in the document]
Comment (patrickweh) [reply src/Rulesets/MailMessage/SendMailRuleset.php:21]: will manage this with array_unique in the action

File: src/Rulesets/MailMessage/SendMailRuleset.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            'subject' => 'nullable|string|max:255',
+            'text_body' => 'nullable|string',
+            'html_body' => 'nullable|string',
+            'attachments' => 'nullable|array',
+            'blade_parameters' => 'nullable',
+            'blade_parameters_serialized' => 'nullable|boolean',
```
Comment (SirSplasch) [inline src/Rulesets/MailMessage/SendMailRuleset.php]: remove nullable

**Review by SirSplasch — APPROVED** [review]
